.buildkite/scripts/steps/esql_grammar_sync.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+cd "$PARENT_DIR" || exit
+
+rm -rf elasticsearch
+git clone https://github.com/elastic/elasticsearch --depth 1 || exit
+
+cd "$KIBANA_DIR" || exit
+
+# Source and destination paths
+source_file="$PARENT_DIR/elasticsearch/x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4"
+destination_file="./packages/kbn-monaco/src/esql/antlr/esql_lexer.g4"
+
+# Copy the file
+cp "$source_file" "$destination_file"
+
+# Replace the line containing "lexer grammar" with "lexer grammar esql_lexer;"
+sed -i -e 's/lexer grammar.*$/lexer grammar esql_lexer;/' "$destination_file" || exit
+
+# Insert "options { caseInsensitive = true; }" one line below
+sed -i -e '/lexer grammar esql_lexer;/a\
+options { caseInsensitive = true; }' "$destination_file" || exit
+
+echo "File copied and modified successfully. Checking for differences."
+
+# Check for differences
+git diff --exit-code --quiet "$destination_file"
+if [ $? -ne 0 ]; then
+  echo "Differences found. Building ANTLR stuff."
+
+  # Built ANTLR stuff
+  sudo bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+  cd ./packages/kbn-monaco/src 
+  yarn build:antlr4
```

Comment:
@drewdaemon we are going to make `antlr` available on CI without installing it using `brew`. So here you will have to run `yarn run build:antlr4:painless && npm run build:antlr4:esql` or add a new script called for example `ci-build` that will not execute `"prebuild:antlr4"` to avoid brew and call it here